test/uk/gov/hmrc/apiscope/repository/ScopeRepositorySpec.scala

```diff
@@ -16,106 +16,96 @@
+   val d = index.toBsonDocument
+   // calling index.remove("v") leaves index untouched - convert to BsonDocument first..
+   d.remove("v") // version
+   d.remove("ns")
+   d
   }
 
   override def beforeEach() {
-    dropRepository(repository)
-    await(repository.ensureIndexes)
-  }
-
-  override protected def afterAll() {
-    dropRepository(repository)
+    super.beforeEach()
+    dropMongoDb()
+    await(repo.ensureIndexes)
   }
 
   "saveScope" should {
     "create scopes and retrieve them from database" in {
-      await(repository.save(scope1))
-      await(repository.save(scope2))
+      await(repo.save(scope1))
+      await(repo.save(scope2))
 
-      await(repository.fetch(scope1.key)).get shouldBe scope1
-      await(repository.fetch(scope2.key)).get shouldBe scope2
+      await(repo.fetch(scope1.key)).get shouldBe scope1
+      await(repo.fetch(scope2.key)).get shouldBe scope2
     }
 
     "update a scope" in {
-      await(repository.save(scope1))
-      await(repository.save(scope2))
+      await(repo.save(scope1))
+      await(repo.save(scope2))
 
       val updatedScope1 = Scope(scope1.key, "updatedName1", "updatedDescription1")
       val updatedScope2 = Scope(scope2.key, "updatedName2", "updatedDescription2", confidenceLevel = Some(L50))
 
-      await(repository.save(updatedScope1))
-      await(repository.save(updatedScope2))
+      await(repo.save(updatedScope1))
+      await(repo.save(updatedScope2))
 
-      await(repository.fetch(scope1.key)).get shouldEqual updatedScope1
-      await(repository.fetch(scope2.key)).get shouldEqual updatedScope2
+      await(repo.fetch(scope1.key)).get shouldEqual updatedScope1
+      await(repo.fetch(scope2.key)).get shouldEqual updatedScope2
     }
   }
 
   "fetchAll" should {
     "retrieve all the scopes from database" in {
 
-      await(repository.save(scope1))
-      await(repository.save(scope2))
+      await(repo.save(scope1))
+      await(repo.save(scope2))
 
-      val allScopes = await(repository.fetchAll())
+      val allScopes = await(repo.fetchAll())
 
-      allScopes should contain theSameElementsAs Seq(scope1, scope2)
+      allScopes should contain allOf(scope1, scope2)
     }
   }
 
   "The indexes in the 'scope' collection" should {
     "have all the indexes" in {
 
-      import scala.concurrent.duration._
-
-      val expectedIndexes = List(
-        Index(key = Seq("key" -> IndexType.Ascending), name = Some("keyIndex"), unique = true, background = true, version = Some(2)),
-        Index(key = Seq("_id" -> IndexType.Ascending), name = Some("_id_"), unique = false, background = false, version = Some(2))
-      )
-
-      val repo: ScopeRepository = createRepository
-
-      eventually(timeout(4.seconds), interval(100.milliseconds)) {
-        getIndexes(repo) shouldBe expectedIndexes
-      }
-
-      dropRepository(repo)
+      val indexes = getIndexes()
+      indexes.size mustEqual 2
+      indexes(1).get("name") mustEqual BsonString("keyIndex")
```

Comment:
We shouldn't rely on the `keyIndex` being at this position in the indexes collection